cmd/minishift/cmd/root.go

```diff
@@ -254,3 +281,43 @@ func performPostUpdateExecution(markerPath string) error {
 
 	return nil
 }
+
+func ensureAllInstanceConfigPath(configPath string) {
+	configDir := filepath.Dir(configPath)
+	if err := os.MkdirAll(configDir, 0777); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", configDir))
+	}
+}
+
+func createMinishiftDirs(dirs *minishiftConfig.MinishiftDirs) {
+	dirPaths := reflect.ValueOf(*dirs)
+
+	for i := 0; i < dirPaths.NumField(); i++ {
+		path := dirPaths.Field(i).Interface().(string)
+		if err := os.MkdirAll(path, 0777); err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating directory: %s", path))
+		}
+	}
+}
+
+// If there is no active profile we need to set minishift as the default profile.
+// Because this will make the profile behaviour backward compatible and consistent with user expectation.
+func setDefaultActiveProfile() {
+	if minishiftConfig.AllInstancesConfig == nil {
+		atexit.ExitWithMessage(1, "Error: All instance config is not initialized")
+	}
+	activeProfile := profileActions.GetActiveProfile()
+	if activeProfile == "" {
+		err := profileActions.SetActiveProfile(constants.DefaultProfileName)
+		if err != nil {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error while setting default profile '%s': %s", profile, err.Error()))
```

Comment:
what is `profile` here pointing at?

```
@@ -51,7 +51,6 @@ import (
 
 const (
        showLibmachineLogs    = "show-libmachine-logs"
+       profile               = "profile"
        enableExperimentalEnv = "MINISHIFT_ENABLE_EXPERIMENTAL"
 )
 ```

Reply:
Fixed it .It should be constants.DefaultProfile.